### PR TITLE
fix(tests): Fix pad pref cookies in tests

### PR DESCRIPTION
**Pull request by orblivion**
This fixes a bug in frontend test helper with setting and clearing of pad prefs cookies. This is something I wrote myself initially because I needed them for ep_webrtc tests.

The problem was caused when pad cookies were switched to a new system. Previously, there was no "path=" for the pad prefs. Now we have `path=/` as well as `sameSite=Lax`. There's also an interesting nuance regarding setting the cookies before vs after the frame is loaded. Beforehand, it was only possible after it was loaded. Now, it's not only possible but necessary to do it before it's loaded. I go into details in a big comment to try to preserve some of the understanding in case this comes up again.

I also updated a couple tests in this PR to properly test these helper functions. None of the other tests in etherpad-lite use the pad settings. ep_webrtc tests make heavy use of them, however, and I've confirmed that this has improved things there. Do we know if any other plugins started using the pad prefs cookies in tests? (If so, the tests are probably broken at the moment)

### CODE REVIEW

**Comment by JohnMcLear** [comment]
Took me a minute to figure out that {delete: "me"} was the actual cookie but looks good to me :)

cc @rhansen

**Review by JohnMcLear — APPROVED** [review]

**Review by rhansen — COMMENTED** [review]
There is a lot going on here that is not easy to understand. Can this commit be split into multiple easier-to-digest commits? Can you explain in the commit message what was wrong before and why?

**Comment by orblivion** [comment]
Oh drat. I didn't have notifications for this repository so I missed the questions.

So, this was for two test helper functions, `addPadPrefCookie` and `setPadPrefCookie`. There was a change in how cookies are done in the main code which broke how these worked in tests.

I created these two functions myself initially for use in ep_webrtc tests. From John's comment, it sounded like my ep_webrtc code would likely be scrapped anyway. As far as I know, these two functions are not used anywhere else (and if they are, they're broken there as well).

Perhaps you'd want to just delete them at this point?

**Comment by orblivion** [comment]
@rhansen :point_up: (in case you didn't get notified here)

File: src/tests/frontend/specs/helper.js [hunk 1]
```diff
@@ -55,76 +55,96 @@ describe('the test helper', function () {
...
       expect(window.document.cookie).to.contain('language=bar');
 
-      helper.newPad(() => {
-        // helper function seems to have cleared cookies
-        // NOTE: this doesn't yet mean it's proven to have taken effect by this point in execution
-        const firstCookie = window.document.cookie;
-        expect(firstCookie).to.not.contain('token=foo');
-        expect(firstCookie).to.not.contain('language=bar');
+      helper.newPad({
+        padPrefs: {delete:"me"},
```
[the remark this answers is not in the document]
Comment (orblivion) [reply src/tests/frontend/specs/helper.js:59]: FWIW I think the idea was that it tests that `clearPadPrefCookie` is called when a new pad is created below, by confirming that this pref does not exist.

**Comment by rhansen** [comment]
PR #4986 made a bunch of changes to cookies. Would you rebase onto latest develop?

**Pull request closed without merging**